roles/dcos_agent_windows/tasks/main.yml

```diff
@@ -8,23 +13,13 @@
     path: '{{ base_windows_dir }}\etc\dcos-version.json'
   register: previous_installation
 
-- name: Windows | Copy dcos-version.json to fact directory
-  win_copy:
-    src: "{{ base_windows_dir }}\\etc\\dcos-version.json"
-    dest: "{{ base_windows_dir }}\\etc\\ansible\\facts.d\\dcos_installation.fact"
-    remote_src: yes
-  when: previous_installation.stat.exists and previous_installation.stat.isreg
-
 - name: Windows | Create dcos_installation.fact.ps1
   win_copy:
-    content: get-content {{ base_windows_dir }}\etc\ansible\facts.d\dcos_installation.fact -Raw | ConvertFrom-Json
+    content: get-content {{ base_windows_dir }}\\etc\\dcos-version.json -Raw | ConvertFrom-Json
     dest: "{{ base_windows_dir }}\\etc\\ansible\\facts.d\\dcos_installation.fact.ps1"
   when: previous_installation.stat.exists and previous_installation.stat.isreg
-
-- name: Windows | Re-run setup to use custom facts
-  setup:
-    fact_path: "{{ base_windows_dir }}\\etc\\ansible\\facts.d\\dcos_installation.fact.ps1"
-  when: previous_installation.stat.exists and previous_installation.stat.isreg
```

Comment:
@fatz , 
Task "Re-run setup to use custom facts" with `setup` module cannot be deleted as it requires `fact_path` argument. Realization of `setup` module is different for Linux vs Windows. You need either leave the task here, or move the block into handlers/main.yml

Reply:
yep indeed... that was an mistake...